packages/expo-location/android/src/main/java/expo/modules/location/LocationModule.java

```diff
@@ -148,35 +150,111 @@ public void onCreate(ModuleRegistry moduleRegistry) {
+        promise.resolve(handleLegacyPermissions(result));
+      }, Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_BACKGROUND_LOCATION);
+    } else {
+      requestForegroundPermissionsAsync(promise);
+    }
+  }
+
+  @Deprecated
+  @ExpoMethod
+  public void getPermissionsAsync(final Promise promise) {
+    if (mPermissionsManager == null) {
+      promise.reject("E_NO_PERMISSIONS", "Permissions module is null. Are you sure all the installed Expo modules are properly linked?");
+      return;
+    }
+
+    if (Build.VERSION.SDK_INT == Build.VERSION_CODES.Q) {
+      mPermissionsManager.getPermissions(result -> {
+        promise.resolve(handleLegacyPermissions(result));
+      }, Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_BACKGROUND_LOCATION);
+    } else {
+      getForegroundPermissionsAsync(promise);
+    }
+  }
+
+  @ExpoMethod
+  public void requestForegroundPermissionsAsync(final Promise promise) {
+    if (mPermissionsManager == null) {
+      promise.reject("E_NO_PERMISSIONS", "Permissions module is null. Are you sure all the installed Expo modules are properly linked?");
+      return;
+    }
     mPermissionsManager.askForPermissions(result -> {
-      promise.resolve(handleLocationPermissions(result));
+      promise.resolve(handleForegroundLocationPermissions(result));
     }, Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION);
   }
 
   @ExpoMethod
-  public void getPermissionsAsync(final Promise promise) {
+  public void requestBackgroundPermissionsAsync(final Promise promise) {
+    if (mPermissionsManager == null) {
+      promise.reject("E_NO_PERMISSIONS", "Permissions module is null. Are you sure all the installed Expo modules are properly linked?");
+      return;
+    }
+
+    if (!isBackgroundPermissionInManifest()) {
+      promise.reject("ERR_NO_PERMISSIONS", "You need to add `ACCESS_BACKGROUND_LOCATION` to the AndroidManifest.");
+      return;
+    }
+
+    if (!shouldAskBackgroundPermissions()) {
+      requestForegroundPermissionsAsync(promise);
+      return;
+    }
+    mPermissionsManager.askForPermissions(result -> {
+      promise.resolve(handleBackgroundLocationPermissions(result));
+    }, Manifest.permission.ACCESS_BACKGROUND_LOCATION);
+  }
+
+  @ExpoMethod
+  public void getForegroundPermissionsAsync(final Promise promise) {
     if (mPermissionsManager == null) {
       promise.reject("E_NO_PERMISSIONS", "Permissions module is null. Are you sure all the installed Expo modules are properly linked?");
       return;
     }
     mPermissionsManager.getPermissions(result -> {
-      promise.resolve(handleLocationPermissions(result));
+      promise.resolve(handleForegroundLocationPermissions(result));
     }, Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION);
```

Comment:
Maybe it's a good idea to make the combination between `ACCESS_FINE_LOCATION` and `ACCESS_COARSE_LOCATION` optional. So devs could use either one too (without defining that in their manifest).

That should cover the usecase of:
- Exclude background location permission
- On android versions before the `ACCESS_BACKGROUND_LOCATION` permission

Reply:
I think is a good idea but we should make those changes in a separate PR.